frontend/packages/design-system/src/components/Icon.tsx

```diff
@@ -1,163 +1,31 @@
-import React from 'react';
-import { Icon as IconNB, NativeBaseProvider } from 'native-base';
-import { Path } from 'react-native-svg';
+import React, { FC } from 'react';
+import { Icon as IconNB, IIconProps } from 'native-base';
 
-import { IconName, IconProps } from '../types/icons';
+import { iconMap } from '../assets/iconMap';
+import { IconName } from '../types/icons';
 
-import * as icons from './icons';
+type Props = {
+  name: keyof typeof iconMap;
+} & Omit<IIconProps, 'name'> & { name: IconName };
```

Comment:
Why do we have `& { name: IconName }`? This is overriding `name: keyof typeof iconMap;`